scim-core/src/main/java/org/apache/directory/scim/core/repository/ObjectMapperProvider.java

```diff
@@ -0,0 +1,27 @@
+package org.apache.directory.scim.core.repository;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.AnnotationIntrospector;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.introspect.AnnotationIntrospectorPair;
+import com.fasterxml.jackson.databind.introspect.JacksonAnnotationIntrospector;
+import com.fasterxml.jackson.module.jakarta.xmlbind.JakartaXmlBindAnnotationIntrospector;
+import com.fasterxml.jackson.module.jakarta.xmlbind.JakartaXmlBindAnnotationModule;
+
+public class ObjectMapperProvider {
```

Comment:
I recently refactored some of the JSON/ObjectMapper bits.
I'm not 100% that will help with this usage, but it is something we can investigate.

https://github.com/apache/directory-scimple/blob/develop/scim-core/src/main/java/org/apache/directory/scim/core/json/ObjectMapperFactory.java 

This _may_ come into play when dealing with patches that contain values from SCIM Extensions.
That said, with your approach using `objectAsMap` it _may_ not make a difference.
(Mostly a note to myself, to go back and look at the ObjectMapper-related bits 😄)

Reply:
of course, feel free to merge :)

Yeah I got the map conversion idea from this pr - https://github.com/apache/directory-scimple/pull/29
which might not be the most efficient but it seems simple enough...